### PR TITLE
Payload classes migration to Lombok

### PR DESCRIPTION
**Pull request by Trimax**
Payload classes were migrated to Lombok

### CODE REVIEW

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/UserActivity.java [hunk 1]
```diff
@@ -2,41 +2,25 @@
...
-            String isoLanguageCode
-    ) {
-        this.guid = guid;
-        this.title = title;
-        this.subtitle = subtitle;
-        this.typeName = typeName;
-        this.status = status;
-        this.isoLanguageCode = isoLanguageCode;
-    }
+    String isoLanguageCode;
```
Comment (bskinner) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/UserActivity.java:23]: Can we add a `@Getter(onMethod = @__(@Override))` to this field and remove the manual getter implementation?

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/medicalprovider/PostPatchMedicalProviderRequestPayload.java [hunk 2]
```diff
@@ -1,55 +1,32 @@
...
-    public String getState() {
-        return state;
-    }
-
     public static class Fields {
         public static final String INSTITUTION_NAME = "institutionName";
         public static final String PHYSICIAN_NAME = "physicianName";
         public static final String CITY = "city";
         public static final String STATE = "state";
     }
```
Comment (bskinner) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/medicalprovider/PostPatchMedicalProviderRequestPayload.java:31]: Is this class referenced elsewhere? If not, let's just move the strings directly into the `SerializedName` annotations above.

**Review by bskinner — COMMENTED** [review]
It's great to see all those manually implemented getters _gone_! Couple of minor changes, then I think we're good to merge!

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/GovernedParticipant.java [hunk 3]
```diff
@@ -1,32 +1,25 @@
...
-        return userGuid;
-    }
-
-    public String getAlias() {
-        return alias;
-    }
-
-    public Profile getUserProfile() {
-        return userProfile;
+        this.userProfile = Optional.ofNullable(userProfile).map(Profile::new).orElse(null);
```
Comment (MocanaAtBroad) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/GovernedParticipant.java:23]: You sure about the constructor in `map` doing the right thing with argument?

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/UserActivity.java (hunk 1 above)
Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/UserActivity.java:23]: I'll just remove the getter itself. It's redundant. I don't think we need to specify the annotation explicitly. The override annotation has retention policy SOURCE, so it doesn't really matter.

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/medicalprovider/PostPatchMedicalProviderRequestPayload.java (hunk 2 above)
Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/medicalprovider/PostPatchMedicalProviderRequestPayload.java:31]: Yep, it is used in the PatchMedicalProviderRoute

**Review by bskinner — APPROVED** [review]
👍

**Review by MocanaAtBroad — COMMENTED** [review]
All looks good. Just saw one that looks suspicious.

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/GovernedParticipant.java (hunk 3 above)
Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/GovernedParticipant.java:23]: I'm sure. It's just a method reference to the class's constructor

Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/json/GovernedParticipant.java:23]: `.map(X::new)` is equivalent to `.map(x -> new X(x))`

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**